platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/collections/LongListDisk.java

```diff
@@ -157,57 +161,43 @@ protected void onEmptyOrAbsentSourceFile(final Path path) throws IOException {
-            MerkleDbFileUtils.completelyWrite(tempFileCHannel, transferBuffer, 0);
-            chunkList.set(firstChunkWithDataIndex, 0L);
-
-            // copy everything except for the first chunk and the last chunk
-            final int numberOfFullChunks = totalNumberOfChunks - firstChunkWithDataIndex - 2;
-            if (numberOfFullChunks > 0) {
-                final long bytesToTransfer = (long) numberOfFullChunks * memoryChunkSize;
-                final long bytesTransferred = MerkleDbFileUtils.completelyTransferFrom(
-                        tempFileCHannel, fileChannel, memoryChunkSize, bytesToTransfer);
-                if (bytesTransferred != bytesToTransfer) {
-                    throw new IOException("Failed to read long list chunks, expected=" + bytesToTransfer + " actual="
-                            + bytesTransferred);
-                }
+            this.tempFileChannel = rf.getChannel();
+            super.readBodyFromFileChannelOnInit(sourceFileName, fileChannel);
+        } finally {
+            if (this.tempFileChannel != null && this.tempFileChannel.isOpen()) {
+                this.tempFileChannel.close();
             }
+        }
+    }
 
-            // copy the last chunk
-            transferBuffer.clear();
-            MerkleDbFileUtils.completelyRead(fileChannel, transferBuffer);
-            transferBuffer.flip();
-            MerkleDbFileUtils.completelyWrite(
-                    tempFileCHannel, transferBuffer, (long) lastChunkWithDataIndex * memoryChunkSize);
+    /** {@inheritDoc} */
+    @Override
+    protected void readChunkData(FileChannel fileChannel, int chunkIndex, int startIndex) throws IOException {
+        final int firstChunkWithDataIndex = toIntExact(minValidIndex.get() / numLongsPerChunk);
+        final long chunkOffset = (long) (chunkIndex - firstChunkWithDataIndex) * memoryChunkSize;
 
-            for (int i = firstChunkWithDataIndex + 1; i < totalNumberOfChunks; i++) {
-                chunkList.set(i, (long) (i - firstChunkWithDataIndex) * memoryChunkSize);
-            }
+        final ByteBuffer transferBuffer = initOrGetTransferBuffer();
+        fillBufferWithZeroes(transferBuffer);
+        transferBuffer.clear();
+
+        final int startByteOffset = startIndex * Long.BYTES;
+        transferBuffer.position(startByteOffset).limit(memoryChunkSize);
+        MerkleDbFileUtils.completelyRead(fileChannel, transferBuffer);
+        transferBuffer.flip();
+        transferBuffer.limit(memoryChunkSize);
+        transferBuffer.position(0);
+        long bytesTransferred = MerkleDbFileUtils.completelyWrite(tempFileChannel, transferBuffer, chunkOffset);
+        if (bytesTransferred != memoryChunkSize) {
+            throw new IOException("Failed to read long list chunks, chunkIndex=" + chunkIndex + " expected="
+                    + memoryChunkSize + " actual=" + bytesTransferred);
         }
+
+        chunkList.set(chunkIndex, chunkOffset);
```

Comment:
This call can be moved to the super class, given that `readChunkData()` returns a newly created chunk.